util/osuser/user.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package osuser implements OS user lookup. It's a wrapper around os/user that
+// works on non-cgo builds.
+package osuser
+
+import (
+	"context"
+	"errors"
+	"log"
+	"os/exec"
+	"os/user"
+	"runtime"
+	"strings"
+	"time"
+	"unicode/utf8"
+
+	"tailscale.com/version/distro"
+)
+
+// LookupUIDAndShell is like os/user.LookupId but handles a few edge cases like
+// gokrazy and non-cgo lookups, and returns the user shell. User shell lookup
+// is best-effort and may be empty.
+func LookupUIDAndShell(uid string) (u *user.User, shell string, err error) {
+	return lookup(uid, user.LookupId, true)
+}
+
+// LookupUsernameAndShell is like os/user.Lookup but handles a few edge cases
+// like gokrazy and non-cgo lookups, and returns the user shell. User shell
+// lookup is best-effort and may be empty.
+func LookupUsernameAndShell(username string) (u *user.User, shell string, err error) {
+	return lookup(username, user.Lookup, true)
+}
+
+// LookupUIDAndShell is like os/user.LookupId but handles a few edge cases like
+// gokrazy and non-cgo lookups.
+func LookupUID(uid string) (*user.User, error) {
+	u, _, err := lookup(uid, user.LookupId, false)
+	return u, err
+}
+
+// LookupUsername is like os/user.Lookup but handles a few edge cases like
+// gokrazy and non-cgo lookups.
+func LookupUsername(username string) (*user.User, error) {
+	u, _, err := lookup(username, user.Lookup, false)
+	return u, err
+}
+
+// lookupStd is either user.Lookup or user.LookupId.
+type lookupStd func(string) (*user.User, error)
+
+func lookup(usernameOrUID string, std lookupStd, wantShell bool) (*user.User, string, error) {
+	if runtime.GOOS != "linux" {
```

Comment:
we want to use getent on any other platforms? FreeBSD at least has this?

Maybe a TODO?

Reply:
Makes sense, left a TODO for now